core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -448,17 +444,16 @@ public String toString() {
         .add("partition", partitionData)
         .add("record_count", recordCount)
         .add("file_size_in_bytes", fileSizeInBytes)
-        .add("column_sizes", columnSizes)
-        .add("value_counts", valueCounts)
-        .add("null_value_counts", nullValueCounts)
-        .add("nan_value_counts", nanValueCounts)
-        .add("lower_bounds", lowerBounds)
-        .add("upper_bounds", upperBounds)
+        .add("column_sizes", toReadableMap(columnSizes))
+        .add("value_counts", toReadableMap(valueCounts))
+        .add("null_value_counts", toReadableMap(nullValueCounts))
+        .add("nan_value_counts", toReadableMap(nanValueCounts))
+        .add("lower_bounds", toReadableMap(lowerBounds))
+        .add("upper_bounds", toReadableMap(upperBounds))
```

Comment:
The original maps should work fine here. There's no risk of `add` modifying the map.